src/components/Review/DecisionPreview/ReviewPreviewModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react'
+import { Modal, Message, Button, Loader, ModalProps } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../../contexts/Localisation'
+import { postRequest } from '../../../utils/helpers/fetchMethods'
+import config from '../../../config'
+import { getItemDisplayComponent, ActionResultPreviewData } from './PreviewItems'
+
+interface PreviewProps extends ModalProps {
+  open: boolean
+  setOpen: (open: boolean) => void
+  decision: string
+  reviewId: number
+  applicationDataOverride: { [key: string]: any }
+}
+
+const previewEndpoint = `${config.serverREST}/preview-actions`
+
+const ReviewPreviewModal: React.FC<PreviewProps> = ({
+  open,
+  setOpen,
+  decision,
+  reviewId,
+  applicationDataOverride,
+}) => {
+  const { strings } = useLanguageProvider()
+  const [data, setData] = useState<ActionResultPreviewData[] | null>(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(false)
+
+  useEffect(() => {
+    if (!open) {
+      setData(null)
+      setError(false)
+      return
+    }
+    setLoading(true)
+    fetchPreviews(reviewId, applicationDataOverride).then((result) => {
+      if (result.error) setError(result.error)
+      else setData(result.displayData)
+      setLoading(false)
+    })
+  }, [open])
+
+  return (
+    <Modal id="preview-modal" open={open} closeOnDimmerClick={false}>
+      <Modal.Header>{strings.REVIEW_DECISION_PREVIEW_HEADER}</Modal.Header>
+      <Modal.Content>
+        {strings.REVIEW_DECISION_PREVIEW_TEXT} <strong>{decision}</strong>
+        {loading && (
+          <Loader active size="huge">
+            {strings.REVIEW_DECISION_PREVIEW_FETCHING}
+          </Loader>
+        )}
+        {error && (
+          <Message
+            error
+            icon="warning sign"
+            header={strings.REVIEW_DECISION_PREVIEW_ERROR_HEADER}
+            content={strings.REVIEW_DECISION_PREVIEW_ERROR_TEXT}
+          />
+        )}
+        <div id="preview-items">
+          {data &&
```

Comment:
stylistic preference again again `{data?.map(() => ) || (<Message info header= strings.REVIEW_DECISION_NO_PREVIEWS_AVAILABLE} />)`

Reply:
That's not quite the same though, is it? We need to check that not only does `data` exist, but it has a length of more than 0. In your version it won't show the "NO PREVIEWS" message if data is an empty array, will it?

Reply:
Ahh ok true could do data?.length > 0 ?